### PR TITLE
Pinning needed to build docs with Python 3.10 or newer

### PR DESCRIPTION
**Pull request by eerovaher**
### Description

Ubuntu and Python versions used by `Read the Docs` are updated. Python 3.10 is incompatible with `sphinx` 3.5, so its latest allowed version had to be bumped down even further.

See https://github.com/astropy/astropy/issues/11725#issuecomment-1231791331 and the preceding comments for more context.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
"Allowed failure" failure is unrelated. Waiting on RTD.

**Comment by pllim** [comment]
Does this really need backporting? Do we really need to build release branches docs with Python 3.10 ? 💭

**Comment by eerovaher** [comment]
I added the backport labels because the last commits that edited the `Read the Docs` configuration were backported, but I don't have any other reasons beyond that.

**Comment by pllim** [comment]
I am a little afraid reducing the pinned version will wreck havoc in LTS. And if we can resolve the real problem before v5.2 and backport that instead, then backport of this is irrelevant.

@saimn , what do you think?

**Review by pllim — APPROVED** [review]
Let's not backport for now. We can always backport if there is a real need. Thanks!

**Comment by pllim** [comment]
Can you pls rebase and see if CI will be green? Thanks!

**Comment by pllim** [comment]
Looks like maybe this warning is failing the build. Not sure where it is coming from.

```
WARNING: while setting up extension sphinx.addnodes: node class 'meta' is already registered, its visitors will be overridden
```

**Comment by eerovaher** [comment]
https://github.com/sphinx-doc/sphinx/issues/9841#issuecomment-966452425 recommends that we either upgrade `sphinx` or downgrade `docutils` to 0.17. I've made a commit to see if pinning `docutils` works.

File: setup.cfg [hunk 1]
```diff
@@ -94,7 +94,8 @@ all =
     pytest>=7.0
     typing_extensions>=3.10.0.1
 docs =
-    sphinx<4
+    sphinx<3.5
+	docutils<0.18  # This should be removed when `sphinx` is unpinned, #11725
```
Comment (pllim) [inline setup.cfg:98]: Does the indentation mean anything?

**Comment by pllim** [comment]
I think Tom R and Stuart M are concerned about the pinning. Lemme ping them for review.

**Comment by astrofrog** [comment]
Is there any need to update RTD to use Python 3.10? (It works fine with 3.9, right?) Why not wait until we have fixed compatibility with Sphinx 4?

**Comment by eerovaher** [comment]
It was a good idea to open this pull request so that we would know for sure what it takes to build the documentation on Python 3.10 without any warnings. But because it turns out that we would have to pin two packages I would tend to agree that merging this pull request is not worth it.

@pllim, you suggested opening this pull request, so if you agree with what has been said then feel free to close it.

**Comment by pllim** [comment]
OK, closing then. Thanks for the work anyway, @eerovaher ! Now, anyone who wants to build doc in Python 3.10 will know what it takes. 😃

**Pull request closed without merging**